cloudify_terminal/tasks.py

```diff
@@ -74,14 +78,20 @@ def run(**kwargs):
                                         log_file_name=log_file_name)
             ctx.logger.info("Will be used: " + ip)
             break
+
+        except socket.timeout as err:
+            ctx.logger.info("Socket timeout retrying... {}"
+                            .format(str(err)))
+
+        except ssh_exception.NoValidConnectionsError as err:
+            ctx.logger.info("NoValidConnectionsError retrying... {}"
+                            .format(str(err)))
+
         except Exception as ex:
-            ctx.logger.info("Can't connect to %s with %s" % (
-                repr(ip), str(ex)
-            ))
+            ctx.logger.info("Can't connect to:{} with exception:{} and type:{}"
+                            .format(repr(ip), str(ex), str(type(ex))))
     else:
-        raise cfy_exc.NonRecoverableError(
-            "please check your ip list"
-        )
+        raise cfy_exc.OperationRetry(message="Please check your ip list")
```

Comment:
I do not like this message. It's bit confusing provided that we're sending logger before. Suggest to remove it.

Reply:
We can replace it to something more userfriendly, maybe "let's try one more time?"